extensions/interactions/ImageClickInput/directives/ImageClickInput.js

```diff
@@ -45,11 +45,41 @@ oppia.directive('oppiaInteractiveImageClickInput', [
             ($attrs.highlightRegionsOnHoverWithValue === 'true');
           $scope.filepath = imageAndRegions.imagePath;
           $scope.imageUrl = '';
-
-          ImagePreloaderService.getImageUrl($scope.filepath)
-            .then(function(objectUrl) {
-              $scope.imageUrl = objectUrl;
-            });
+          $scope.loadingIndicatorUrl = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/activity/loadingIndicator.gif')
+          );
+          $scope.isLoadingIndicatorShown = false;
+          if (ImagePreloaderService.inExplorationPlayer()) {
```

Comment:
Can we show the loading even in other places than exploration player?

Reply:
Yeah, ImageClickInput.js is for displaying the image in case of ImageAndRegion interaction. 
inExploration is mainly for checking that the imagePreloaderService has started or not.(image preloading has started or not)

Reply:
No, we are not supposed to show loading indicator while editing the exploration. Therefore, this condition helps in differentiating between the two.(player and editor mode).

If we decide it to show even while editing the exploration, then we would not need the condition.
What do you say, shall we show it while editing the exploration ?

Reply:
I think we can show the indicator even in the editor.

Reply:
Okay

Reply:
We can't actually do that, because the new version of the exploration won't be saved by then. So, we can't get the dimensions of the images for showing them in the editor.
We get the dimensions of the image using the ImagePreloaderService's function _geDimensionsOfImage which uses the states of the exploration and that isn't assigned the value until the ImagePreloaderService's init function is called.

Reply:
Ok